packages/zoe/src/objArrayConversion.js

```diff
@@ -59,7 +69,9 @@ export const filterObj = (obj, subsetKeys) => {
     );
     newObj[key] = obj[key];
   });
-  return newObj;
+
+  const picked = /** @type {Pick<T,U[number]>} */ (newObj);
+  return picked;
```

Comment:
```suggestion
  return Object.fromEntries(subsetKeys.map(key => [key, obj[key]]));
```
is almost adequate, except for the error test which, I agree, is good to keep.

Reply:
removed function